File: app/src/main/java/com/hazz/aipick/ui/adapter/MarketsPagerItemAdapter.java
package com.hazz.aipick.ui.adapter;

import android.content.Context;
import android.support.annotation.Nullable;
import android.util.Log;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.hazz.aipick.R;
import com.hazz.aipick.mvp.model.bean.Coin;
import com.hazz.aipick.mvp.model.bean.MarketItem;
import com.hazz.aipick.utils.BigDecimalUtil;

import java.math.BigDecimal;
import java.math.RoundingMode;
import java.util.List;

public class MarketsPagerItemAdapter extends BaseQuickAdapter<Coin.DataBean, BaseViewHolder> {

    //    private String formatPercent;
    public MarketsPagerItemAdapter(Context context, int layoutResId, @Nullable List<Coin.DataBean> data) {
        super(layoutResId, data);
    }


    @Override
    protected void convert(BaseViewHolder helper, Coin.DataBean marketItem) {
        if (marketItem == null) {
            return;
        }
      Log.d("ceshi",marketItem.priceUsd);
//
      String changeString = marketItem.volumePercent;
        String close = marketItem.priceUsd;
        String volume = marketItem.volumeUsd24Hr;
//        String cnyPrice = marketItem.cnyPrice;
        String cyn= BigDecimalUtil.mul(close,"7",2);
        helper
                .setText(R.id.tv_trade_a, marketItem.baseSymbol)
                .setText(R.id.tv_trade_b, marketItem.quoteSymbol)
                .setText(R.id.tv_price, close == null ? "--" : close)
                .setText(R.id.tv_volume, (volume == null || volume.equals("--")) ? "--" : new BigDecimal(volume).setScale(0, RoundingMode.HALF_UP).toPlainString())
                .setText(R.id.up_down, changeString == null ? "0.00%" : changeString)
                .setText(R.id.tv_price_legal, "¥ " + (cyn.equals("0") ? "--" : cyn));
//                .setBackgroundRes(R.id.up_down, marketItem.isUp ? R.drawable.shape_oval_green_3 : R.drawable.shape_oval_red_3);
////        if("USDT".equals(marketItem.tradeB) && !TextUtils.isEmpty(close)){
////            //Log.d("YUDAN", "convert: change to ￥");
////            String res = new DecimalFormat("########.####").format( 6.9*Float.valueOf(close) );
////            helper.setText(R.id.tv_price_legal,"¥ "+res);
////        }
    }
}

File: app/src/main/java/com/hazz/aipick/ui/adapter/MarketsPagerItemAdapter.kt
package com.hazz.aipick.ui.adapter

import android.content.Intent
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.hazz.aipick.R
import com.hazz.aipick.socket.CoinDetail
import com.hazz.aipick.ui.activity.CoinDescActivity
import com.hazz.aipick.utils.BigDecimalUtil
import com.hazz.aipick.utils.SPUtil
import com.orhanobut.logger.Logger

class MarketsPagerItemAdapter//    private String formatPercent;
(layoutResId: Int, data: List<CoinDetail>?) : BaseQuickAdapter<CoinDetail, BaseViewHolder>(layoutResId, data) {

    private var sortByPrice: Boolean = false
    private var sortByUp: Boolean = false
    fun sortByPrice() {
        if (sortByPrice) {
            data.sortBy { BigDecimalUtil.mul(it.tick.close, "1").toDouble() }
        } else {
            data.sortByDescending { BigDecimalUtil.mul(it.tick.close, "1").toDouble() }
        }
        sortByPrice = !sortByPrice
        notifyDataSetChanged()
    }

    private var market: String = "--"
    fun setMarket(market: String) {
        this.market = market
    }

    fun sortByUp() {
        if (sortByUp) {
            data.sortBy {
                BigDecimalUtil.mulByDecimal(
                        BigDecimalUtil.div(BigDecimalUtil.sub(it.tick.close, it.tick.open, 6), it.tick.open, 6),
                        "1").toDouble()
            }
        } else {
            data.sortByDescending {
                BigDecimalUtil.mulByDecimal(
                        BigDecimalUtil.div(BigDecimalUtil.sub(it.tick.close, it.tick.open, 6), it.tick.open, 6),
                        "1").toDouble()
            }
        }
        sortByUp = !sortByUp
        notifyDataSetChanged()
    }


    override fun convert(helper: BaseViewHolder, marketItem: CoinDetail?) {
        if (marketItem == null) {
            return
        }
        Logger.t("test").d(marketItem)
        var close = ""
        val tick = marketItem.tick ?: return
//String changeString = marketItem.volumePercent;
        if (tick.close != null) {
            close = tick.close
        }
        var volume = ""
        if (tick.vol != null) {
            volume = tick.vol
        }
        val ch = marketItem.ch
        val split = ch.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val coinName = split[1].substring(0, split[1].length - 4)

        val compare = BigDecimalUtil.compare(tick.close, tick.open)
        marketItem.isUp = compare != -1

        val rate = BigDecimalUtil.mul(BigDecimalUtil.div(BigDecimalUtil.sub(tick.close, tick.open, 6), tick.open, 6), "100", 2) + "%"

        //        String cnyPrice = marketItem.cnyPrice;
        val rate1 = SPUtil.getRate()
        val cny = BigDecimalUtil.mul(close, rate1, 4)
        helper
                .setText(R.id.tv_trade_a, coinName.toUpperCase())
                .setText(R.id.tv_price, if (cny == "0") "--" else cny)
                .setText(R.id.tv_num_tip, market)
                .setText(R.id.up_down, if (marketItem.isUp) "+$rate" else rate)
                .setText(R.id.tv_price_legal, "$ " + BigDecimalUtil.format(close, 4))
                .setText(R.id.trade_amount, if (volume == "--") "--" else BigDecimalUtil.format(volume, 2))
                .setTextColor(R.id.up_down, if (marketItem.isUp) mContext.resources.getColor(R.color.main_color_green) else mContext.resources.getColor(R.color.main_color_red))

        helper.itemView.setOnClickListener {
            mContext.startActivity(Intent(mContext, CoinDescActivity::class.java).putExtra("name", marketItem.ch).putExtra("market", market))
        }
    }
}
